src/test/java/net/minecraftforge/debug/client/LayerRenderTypeTest.java

```diff
@@ -0,0 +1,147 @@
+            ITEMS.register(FMLJavaModLoadingContext.get().getModEventBus());
+        }
+    }
+
+    @Mod.EventBusSubscriber(modid = MOD_ID, value = Dist.CLIENT, bus = Mod.EventBusSubscriber.Bus.MOD)
+    private static class ClientHandler
+    {
+        private static ShaderInstance CUSTOM_SHADER;
+
+        private static final ResourceLocation CUSTOM_RENDER_TYPE_NAME = new ResourceLocation(MOD_ID, "custom");
+        private static RenderType CUSTOM_RENDER_TYPE;
+
+        @SubscribeEvent
+        public static void onClientSetup(final FMLClientSetupEvent event)
+        {
+            if (!ENABLED) return;
+
+            ItemBlockRenderTypes.setRenderLayer(TEST_BLOCK.get(), CUSTOM_RENDER_TYPE);
+        }
+
+        @SubscribeEvent
+        public static void onRegisterRenderTypes(RenderTypeRegisterEvent event)
+        {
+            if (!ENABLED) return;
+
+            CUSTOM_RENDER_TYPE = ComplexRenderType.builder()
+                    .beforeChunkRenderCallback(ClientHandler::beforeChunkRender)
+                    .build(
+                            CUSTOM_RENDER_TYPE_NAME.toString(),
+                            DefaultVertexFormat.BLOCK, VertexFormat.Mode.QUADS,
+                            RenderType.SMALL_BUFFER_SIZE, true, false,
+                            RenderType.CompositeState.builder()
+                                    .setLightmapState(new RenderStateShard.LightmapStateShard(true))
+                                    .setShaderState(new RenderStateShard.ShaderStateShard(() -> CUSTOM_SHADER))
+                                    .setTextureState(new RenderStateShard.TextureStateShard(TextureAtlas.LOCATION_BLOCKS, false, false))
+                                    .createCompositeState(true)
+                    );
+        }
+
+        @SubscribeEvent
+        public static void onRegisterSolidStaticRenderTypes(LayerRenderTypeRegisterEvent.Solid event)
+        {
+            if (!ENABLED) return;
+
+            event.getBuilder().add(
+                    CUSTOM_RENDER_TYPE, CUSTOM_RENDER_TYPE_NAME,
+                    List.of(RenderType.cutout()), // After cutout
+                    List.of()
+            );
+        }
+
+        @SubscribeEvent
+        public static void onRegisterShaders(RegisterShadersEvent event) throws IOException
+        {
+            if (!ENABLED) return;
+
+            event.registerShader(
+                    new ShaderInstance(event.getResourceManager(), new ResourceLocation(MOD_ID, "rendertype_custom"), DefaultVertexFormat.BLOCK),
+                    shader -> CUSTOM_SHADER = shader
+            );
+        }
+
+        private static void beforeChunkRender(
+            LevelRenderer levelRenderer,
+            ChunkRenderDispatcher.RenderChunk renderChunk,
+            PoseStack poseStack,
+            double cameraX,
+            double cameraY,
+            double cameraZ,
+            Matrix4f projectionMatrix
+        ) {
```

Comment:
Formatting: braces on newlines:
```suggestion
        )
        {
```